src/linear_algebra/alternating.lean

```diff
@@ -0,0 +1,239 @@
+
+@[ext] theorem ext {f f' : alternating_map R M N ι} (H : ∀ x, f x = f' x) : f = f' :=
+coe_inj (funext H)
+
+theorem ext_iff {f g : alternating_map R M N ι} : f = g ↔ ∀ x, f x = g x :=
+⟨λ h x, h ▸ rfl, λ h, ext h⟩
+
+instance : has_coe (alternating_map R M N ι) (multilinear_map R (λ i : ι, M) N) :=
+⟨λ x, x.to_multilinear_map⟩
+
+@[simp, norm_cast] lemma coe_multilinear_map : ⇑(f : multilinear_map R (λ i : ι, M) N) = f := rfl
+
+@[simp] lemma to_multilinear_map_eq_coe : f.to_multilinear_map = f := rfl
+
+@[simp] lemma coe_multilinear_map_mk (f : (ι → M) → N) (h₁ h₂ h₃) :
+  ((⟨f, h₁, h₂, h₃⟩ : alternating_map R M N ι) :  multilinear_map R (λ i : ι, M) N) = ⟨f, h₁, h₂⟩ :=
+rfl
+
+end coercions
+
+/-!
+### Simp-normal forms of the structure fields
+
+These are expressed in terms of `⇑f` instead of `f.to_fun`.
+-/
+@[simp] lemma map_add (i : ι) (x y : M) :
+  f (update v i (x + y)) = f (update v i x) + f (update v i y) :=
+f.to_multilinear_map.map_add' v i x y
+
+@[simp] lemma map_sub (i : ι) (x y : M') :
+  g' (update v' i (x - y)) = g' (update v' i x) - g' (update v' i y) :=
+g'.to_multilinear_map.map_sub v' i x y
+
+@[simp] lemma map_smul (i : ι) (r : R) (x : M) :
+  f (update v i (r • x)) = r • f (update v i x) :=
+f.to_multilinear_map.map_smul' v i r x
+
+@[simp] lemma map_eq_args (v : ι → M) {i j : ι} (h : v i = v j) (hij : i ≠ j) :
+  f v = 0 :=
+f.map_eq_args' v i j h hij
+
+/-!
+### Algebraic structure inherited from `multilinear_map`
+
+`alternating_map` carries the same `add_comm_monoid`, `add_comm_group`, and `semimodule` structure
+as `multilinear_map`
+-/
+
+instance : has_add (alternating_map R M N ι) :=
+⟨λ a b,
+  { map_eq_args' := λ v i j h hij, by simp [a.map_eq_args v h hij, b.map_eq_args v h hij],
+    ..(a + b : multilinear_map R (λ i : ι, M) N)}⟩
+
+@[simp] lemma add_apply : (f + f') v = f v + f' v := rfl
+
+instance : has_zero (alternating_map R M N ι) :=
+⟨{map_eq_args' := λ v i j h hij, by simp,
+  ..(0 : multilinear_map R (λ i : ι, M) N)}⟩
+
+@[simp] lemma zero_apply : (0 : alternating_map R M N ι) v = 0 := rfl
+
+instance : inhabited (alternating_map R M N ι) := ⟨0⟩
+
+instance : add_comm_monoid (alternating_map R M N ι) :=
+by refine {zero := 0, add := (+), ..};
+   intros; ext; simp [add_comm, add_left_comm]
+
+instance : has_neg (alternating_map R' M' N' ι) :=
+⟨λ f, {
+  map_eq_args' := λ v i j h hij, by simp [f.map_eq_args v h hij],
+  ..(-(f : multilinear_map R' (λ i : ι, M') N')) }⟩
+
+@[simp] lemma neg_apply (m : ι → M') : (-g') m = - (g' m) := rfl
+
+instance : add_comm_group (alternating_map R' M' N' ι) :=
+by refine {zero := 0, add := (+), neg := has_neg.neg, ..alternating_map.add_comm_monoid, ..};
+   intros; ext; simp [add_comm, add_left_comm]
+
+section semimodule
+
+variables {S : Type*} [comm_semiring S] [algebra S R] [semimodule S N]
+  [is_scalar_tower S R N]
+
+instance : has_scalar S (alternating_map R M N ι) :=
+⟨λ c f, {
+  map_eq_args' := λ v i j h hij, by simp [f.map_eq_args v h hij],
```

Comment:
```suggestion
⟨λ c f,
{ map_eq_args' := λ v i j h hij, by simp [f.map_eq_args v h hij],
```

Reply:
That indentation feels weird to me, since two brackets at different levels of nesting end up at the same indentation...

Reply:
I instead did this how I did `has_add` above